enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -2601,10 +2611,6 @@ func (e *partitionEvictor) generateSamplesForEviction(quitChan chan struct{}) er
 			iter = newIter
 		}
 		totalCount += 1
-		if totalCount > e.samplesPerBatch {
```

Comment:
I was thinking that the timer-based recreation would be a defensive check and that we would still keep the count-based check as well. Did you have any opinion on whether we should keep both or only keep the time-based one?

Reply:
I kept both.

Reply:
The count-based check is still missing.

Reply:
Thanks!..I didn't catch that when I try to recover from a bad git restore command, and I was wondering why the test became flaky...